### PR TITLE
[Bugfix] Fix zero number of rows in the query result

### PR DESCRIPTION
**Pull request by s7monk**
fix: https://github.com/apache/paimon-webui/issues/487

### Purpose

Fix the problem that the number of rows in the query result is 0.

微信截图_20240709182735

### CODE REVIEW

File: paimon-web-ui/src/store/job/index.ts [hunk 1]
```diff
@@ -79,10 +79,10 @@ export const useJobStore = defineStore({
       return (key) => {
         const initResultData = state.jobs[key]?.job?.resultData
         const refreshedResultData = state.jobs[key]?.jobResultData?.resultData
-        if (initResultData)
+        if (initResultData && initResultData.length > 0)
```
Comment (SteNicholas) [inline paimon-web-ui/src/store/job/index.ts:82]: I don't think this should add this condition.

Comment (s7monk) [reply paimon-web-ui/src/store/job/index.ts:82]: If this condition is not added, the result is not as expected, and the rows of the query result data are 0.

Comment (SteNicholas) [reply paimon-web-ui/src/store/job/index.ts:82]: @s7monk, could you explain the reason why need to add this condition?

Comment (s7monk) [reply paimon-web-ui/src/store/job/index.ts:82]: The type of initResultData is ResultDataItem[] | undefined. If only if (initResultData) can only judge the case where it is not undefined, this branch will be entered when initResultData is an empty array, so initResultData.length is returned, which is exactly 0. This is my understanding.

image

**Review by SteNicholas — APPROVED** [review]

**Pull request merged**